parser/Node.h

```diff
@@ -46,7 +92,16 @@ template <class To> To *cast_node(Node *what) {
 #else
     static_assert(false);
 #endif
-    return fast_cast<Node, To>(what);
+
+    if (auto casted = fast_cast<Node, To>(what)) {
+        return casted;
+    }
+
+    // if (auto casted = fast_cast<Node, NodeWithExpr>(what)) {
+    //     return cast_node<To>(casted->wrappedNode.get());
+    // }
```

Comment:
We could tweak `cast_to` like this, to fake it so that `NodeWithExpr` can be casted as if it were its `wrappedNode`.

This solves the immediate issue to make `cast_node<parser::Self>` work, but introduces a new one: the `[&](parser::NodeWithExpr *nodeWithExpr)` lambda in the desugar's big `typecase` won't get hit (because `[&](parser::Self *self)` would get hit, first).